registry/src/registry_helper.rs

```diff
@@ -67,82 +66,107 @@ impl RegistryHelper {
                 key_path: self.config.key_path.clone(),
                 value_name: Some(value_name.clone()),
                 value_data: Some(convert_reg_value(&value)?),
-                exist: None,
+                ..Default::default()
             })
         } else {
             Ok(Registry {
                 key_path: self.config.key_path.clone(),
-                value_name: None,
-                value_data: None,
-                exist: None,
+                ..Default::default()
             })
         }
     }
 
-    pub fn set(&self) -> Result<(), RegistryError> {
+    pub fn set(&self, is_what_if: bool) -> Result<Option<Registry>, RegistryError> {
         let reg_key = match self.open(Security::Write) {
-            Ok((reg_key, _subkey)) => reg_key,
+            Ok((reg_key, _subkey)) => Some(reg_key),
             // handle NotFound error
             Err(RegistryError::RegistryKeyNotFound(_)) => {
                 // if the key doesn't exist, some of the parent keys may
                 // not exist either, so we need to find the valid parent key
                 // and then create the subkeys that don't exist
-                let (parent_key, subkeys) = self.get_valid_parent_key_and_subkeys()?;
-                let mut reg_key = parent_key;
-                for subkey in subkeys {
-                    let Ok(path) = UCString::<u16>::from_str(subkey) else {
-                        return Err(RegistryError::Utf16Conversion("subkey".to_string()));
-                    };
-
-                    reg_key = reg_key.create(path, Security::CreateSubKey)?;
+                if is_what_if {
+                    None
```

Comment:
Do we want to return a message that this key would be created?  Maybe the messages should be an array of strings.